editors/vscode/src/browser.ts

```diff
@@ -6,11 +6,25 @@
 import { Uri } from "vscode";
 import * as vscode from "vscode";
 import { LanguageClientOptions } from "vscode-languageclient";
-import { LanguageClient } from "vscode-languageclient/browser";
+import {
+    HandleWorkDoneProgressSignature,
+    ProgressToken,
+    WorkDoneProgressBegin,
+    WorkDoneProgressEnd,
+    WorkDoneProgressReport,
+    LanguageClient,
+} from "vscode-languageclient/browser";
 import { set_client, PropertiesViewProvider } from "./common";
 
 let client: LanguageClient;
 let statusBar: vscode.StatusBarItem;
+let properties_provider: PropertiesViewProvider;
+
+function extract_uri_from_progress_message(input: string): string {
+    const start = input.indexOf(": ");
+    const end = input.lastIndexOf("@");
+    return input.slice(start + 2, end);
+}
```

Comment:
I suggest to move this function into `common.ts`, which is already shared between `browser.ts` and `extension.ts`.